snuba/datasets/errors_replacer.py

```diff
@@ -420,7 +423,16 @@ def process_delete_tag(
     select_columns = []
     for col in all_columns:
         if is_promoted and col.flattened == tag_column_name:
-            select_columns.append("NULL")
+            # The promoted tag columns of events are non nullable, but those of
+            # errors are non nullable. We check the column against the schema
+            # to determine whether to write an empty string or NULL.
+            column_type = schema.get_data_source().get_columns().get(tag_column_name)
+            assert column_type is not None
+            is_nullable = column_type.type.has_modifier(Nullable)
+            if is_nullable:
+                select_columns.append("NULL")
+            else:
+                select_columns.append("''")
```

Comment:
This is only for tags and not for contexts, isn't it? 
While promoted tags are all strings, the promoted contexts can be integers.

Reply:
Yeah, I don't think we provide the ability to delete a specific context by key like we do with tags.